src/modules/gameDevs/gameDevWebhookHandler.ts

```diff
@@ -16,12 +20,25 @@ import {
 export const getAchievementsThruWebhookHandler: Lifecycle.Method = async (request, h) => {
   const { gameType } = request.pre.webhookValidation;
   const achievements = await findAllAchievementsByGameType(gameType.id);
-  // if () {
-  //   throw Boom.forbidden('User is not the owner of the game');
-  // }
-  return h
-    .response({ achievements: achievements.map((achievement) => achievement.toJSON()) })
-    .code(200);
+
+  return h.response(arrayToJSON(achievements)).code(200);
+};
+
+export const postAchievementsProgressHandler: Lifecycle.Method = async (request, h) => {
+  const { gameType } = request.pre.webhookValidation;
+  const payload = request.pre.webhookPayload as AchievementUnlockedCreationAttributes;
+  const user = request.pre.appendUserToRequest as User;
+
+  const achievement = await validateAchievement(request.params.achievementId, gameType.id);
+
+  const [rslt] = await createOrUpdateAchievementUnlocked({
+    progress: payload.progress,
+    isUnlocked: payload.progress >= achievement.targetValue,
+    _userId: user.id,
+    _achievementId: achievement.id,
+  });
```

Comment:
isUnlocked will be set to true if achievement.targetValue is reached

Reply:
For now, it will be set to true if the client requests so because we are not going to use the target value for our first iteration on Achievements. But in the next iteration, yes, it will work as you said.

Reply:
changed the schema based and now set unlocked to always be true